src/main/java/dk/kb/present/copyright/CopyrightAccessExtractor.java

```diff
@@ -0,0 +1,364 @@
+                }
+                personList.add(person);
+            }
+        }
+        return personList;
+    }
+    
+  /*
+  <cdl:creator>
+    <dk:creator.corporate>
+      <cdl:name>Em. Bærentzen &amp; Co. lith. Inst.</cdl:name>
+      <dk:year.started>1837</dk:year.started>
+      <dk:year.ended>1874</dk:year.ended>
+    </dk:creator.corporate>
+  </cdl:creator>
+  */     
+     public static ArrayList<CreatorCorporate> buildCooperate( Element accessConditionElement){       
+         ArrayList<CreatorCorporate> cooperateList = new  ArrayList<CreatorCorporate>();
+
+         NodeList creatorCorporate = accessConditionElement.getElementsByTagName("dk:creator.corporate");
+         
+         if (creatorCorporate != null) {
+
+             for (int i =0;i<creatorCorporate.getLength();i++) {
+                 CreatorCorporate coorporate=new CopyrightAccessDto().new CreatorCorporate();
+                 Element creatorElement = (Element) creatorCorporate.item(i);
+
+                 String name= creatorElement.getElementsByTagName("cdl:name").item(0).getTextContent();
+                 coorporate.setName(name);
+                 
+                 NodeList yearStartedNode = creatorElement.getElementsByTagName("dk:year.started");
+                 NodeList yearEndedNode = creatorElement.getElementsByTagName("dk:year.ended");
+
+                 if (yearStartedNode.getLength() >0  ){
+                     String yearStarted= yearStartedNode.item(0).getTextContent();                         
+                     coorporate.setYearStarted(yearStarted);
+                 }
+                 if (yearEndedNode.getLength() > 0){
+                     String yearEnded= yearEndedNode.item(0).getTextContent();                         
+                     coorporate.setYearEnded(yearEnded);
+                 }                                  
+                 
+                 cooperateList.add(coorporate);
+             }
+         }
+         return cooperateList;
+     }
+   
+    public static  Document createDocFromXml(String xml) throws Exception{
+
+        //System.out.println(response);
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+
+        XMLEscapeSanitiser sanitiser = new XMLEscapeSanitiser(""); //Do not replace with anything
+        String responseSanitized  =  sanitiser.apply(xml);
+
+        Document document = null;
+        try {
+            document = builder.parse(new InputSource(new StringReader(responseSanitized)));
+            document.getDocumentElement().normalize();
+        }
+        catch(Exception e) {                       
+            log.error("Invalid XML",e);                                                
+            throw new Exception("invalid xml",e);            
+
+        }
+
+        return document;
+    }
+        
+    private static String getImageLink(Document doc) {
```

Comment:
This really does not belong here. Could you add a comment on this being a temporary hack that will be replaced by a more solid solution at a later point?

Reply:
fixed